go/libkb/upak_loader.go

```diff
@@ -108,14 +111,22 @@ func (u *CachedUPAKLoader) getCachedUPAK(ctx context.Context, uid keybase1.UID,
 	} else {
 		var tmp keybase1.UserPlusKeysV2AllIncarnations
 		found, err := u.G().LocalDb.GetInto(&tmp, culDBKeyV2(uid))
+		hit := false
 		if err != nil {
 			u.G().Log.CWarningf(ctx, "trouble accessing UserPlusKeysV2AllIncarnations cache: %s", err)
 		} else if !found {
 			u.G().VDL.CLogf(ctx, VLog0, "| missed disk cache")
-		} else if tmp.MinorVersion != UPK2MinorVersionCurrent {
-			u.G().VDL.CLogf(ctx, VLog0, "| found old minor version %d, but wanted %d; will overwrite with fresh UPAK", tmp.MinorVersion, UPK2MinorVersionCurrent)
+		} else if tmp.MinorVersion == UPK2MinorVersionCurrent {
+			u.G().VDL.CLogf(ctx, VLog0, "| hit disk cache (v%d)", tmp.MinorVersion)
+			hit = true
+		} else if UPK2MinorVersionCurrent == keybase1.UPK2MinorVersion_V6 && tmp.MinorVersion == keybase1.UPK2MinorVersion_V5 && len(tmp.PastIncarnations) == 0 {
```

Comment:
An alternative might be to do a doctoring step, where as we unmarshall a v5 link, we bump it to v6 if there are no resets. That might be slightly less fussy? Not a big difference though either way.

Reply:
hmm, that's a nice solution...